src/Extraload/Extractor/Doctrine/PreparedQueryExtractor.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Extraload\Extractor\Doctrine;
+
+use Doctrine\DBAL\Connection;
+use Extraload\Extractor\ExtractorInterface;
+
+class PreparedQueryExtractor implements ExtractorInterface
+{
+    private $position = 0;
+
+    private $data;
+
+    public function __construct(Connection $conn, string $sql, array $values)
+    {
+        $this->position = 0;
+
+        $stmt = $conn->prepare($sql);
+
+        foreach ($values as $value) {
+            $stmt->bindValue(
+                $value['parameter'],
+                $value['value'],
+                $value['data_type'] ?? null
+            );
+        }
+
+        $stmt->execute();
+
+        $this->data = $stmt->fetchAll();
```

Comment:
For large data sets, and that's common in ETL, fetching all at once can be impossible to process and lead to out of memory issues, especially if doing heavy transformations. It's better if we can extract row by row, and only keep one row in memory at a time.